### PR TITLE
Leaderboard: hit the API 🎯

### PR DESCRIPTION
**Pull request by ritikarawat220**
### Send and receive data from API

- [x]     Add _getScores_ and _postScores_ functions.
- [x]     Implement the _"Refresh"_ button.
- [x]     Implement the form "_Submit_" button.

### CODE REVIEW

File: src/request.js [hunk 1]
```diff
@@ -0,0 +1,20 @@
+const getScores = async (id) => { //eslint-disable-line no-unused-vars
+    const url = "https://us-central1-js-capstone-backend.cloudfunctions.net/api/games/ubc1A0KC5FEKLnzaXFow/scores";
+    const response = await fetch(url, { method: "GET" });
+    const scores = await response.json();
+    return scores;
+  };
```
Comment (Olamarx) [inline src/request.js]: - [ ] I see you are not using the id parameter, it is best practice to remove it. I removed it and it still works fine.

File: src/request.js [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+  
+  const postScores = async (id, body) => {
```
Comment (Olamarx) [inline src/request.js]: - [ ] Same thing applies to this id.

File: src/index.js [hunk 3]
```diff
@@ -1,27 +1,29 @@
...
+    user: document.getElementById("name-input").value,
+    score: parseInt(document.getElementById("score-input").value, 10),
+  };
+  postScores(TEST_GAME_ID, body);
+  document.getElementById("name-input").value = "";
+  document.getElementById("score-input").value = "";
+});
+
+getScores(TEST_GAME_ID).then((scores) => {
+  itemList(scores.result);
```
Comment (Olamarx) [inline src/index.js]: - [ ] It is of best practice to destructure your data for easy usage just as I did in the attached image.
image

File: src/index.js [hunk 4]
```diff
@@ -1,27 +1,29 @@
...
-];
+const TEST_GAME_ID = "ubc1A0KC5FEKLnzaXFow";
 
-itemList(placeholders);
\ No newline at end of file
+const refreshButton = document.getElementById("refresh-btn");
+const form = document.getElementById("form");
+
+refreshButton.addEventListener("click", () => {
+  getScores(TEST_GAME_ID).then((scores) => {
```
Comment (Olamarx) [inline src/index.js]: - [ ] Don't forget to remove the parameter in the getScores.

**Review by Olamarx — CHANGES_REQUESTED** [review]
Hi @ritikarawat220, 

Good job so far, indeed you have spent time on this project!
There are some issues that you still need to work on to go to the next project but you are almost there!


## Highlights
- Linters passed ✔️
- Readme file ✔️

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by ghost — APPROVED** [review]
Hi @ritikarawat220 , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



## Highlights

✅ Correct use of Gitflow
✅ No linter error
✅ Good documentation👍 

## Optional suggestions

N/A

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/index.js (hunk 3 above)
Comment (ritikarawat220) [reply src/index.js]: thankyou for giving good suggestions. 😄

**Comment by ritikarawat220** [comment]
Thanks for Approving! 😄

**Pull request merged**